src/components/common/Dialogs/ControlSafeDialog/validation.ts

```diff
@@ -149,9 +149,8 @@ export const getValidationSchema = (
                         }),
                       });
                     }
-                    const selectedToken = this.parent.tokenData?.tokenAddress;
-                    const selectedTokenDecimals =
-                      this.parent.tokenData?.decimals;
+                    const selectedToken = this.parent.token?.tokenAddress;
+                    const selectedTokenDecimals = this.parent.token?.decimals;
```

Comment:
I also noticed and fixed this in the `improve token storage` PR :smile:

Reply:
Great minds think alike 🧠